components/support/utils/src/main/java/mozilla/components/support/utils/URLStringUtils.kt

```diff
@@ -90,8 +90,9 @@ object URLStringUtils {
         // -x.com
         // http://www-.com
         // www.c-c-
+        // https://disposable-wide4--solidsoft-blog-disposable.netlify.app/ (fenix/issues/10747)
```

Comment:
Let's not include the Fenix reference, as we already have a reference to an Android component issue in the commit message.

This `url` is added below the "There are some examples of non-URLs according to this pattern:",
Instead of adding the whole `url`, I think it would be better if we describe, what this new quantifier is doing.

Like the above comment:
```

 0 or more pairs of consecutive word letters followed by a dash
        //        ---                     ---
        // followed by at least a single word letter.
        // -----------             ----------
        // Combined, that means "w", "w-w", "w-w-w", etc match, but "w-", "w-w-", "w-w-w-" do not.
        //          --------------
        // That surrounds :, :// or .
        //                                                    -
        // At the end, there may be an optional
        //                                    ------------
        // non-word, non-- but still non-space character (e.g., ':', '/', '.', '?' but not 'a', '-', '\t')
```

Reply:
Yes, the regex in the comment above needs an update to reflect the change:

`(\w+-+)*\w+(://[/]*|:|\.)(\w+-+)*\w+([\S&&[^\w-]]\S*)?`  (both groups updated -> see comment below)

`// 0 or more pairs of consecutive word letters followed by a dash` -> This should be come `// 0 or more pairs of consecutive word letters or dashes`